### PR TITLE
cyrusdb_twoskip: use xsyslog for errors

### PR DESCRIPTION
**Pull request by elliefm**
More of this.  Please try to build it, in case your compiler finds stuff mine doesn't.

There's a couple of log lines where I'm not sure what the variables being logged represent, which makes it hard to name them well.  I've annotated these with XXX comments, guidance would be appreciated!

### CODE REVIEW

**Comment by rsto** [comment]
Builds cleanly for me with gcc 8.3.0

File: lib/cyrusdb_twoskip.c [hunk 1]
```diff
@@ -2111,8 +2118,11 @@ static int myconsistent(struct dbengine *db, struct txn *tid)
         for (i = 0; i < record.level; i++) {
             /* check the old pointer was to here */
             if (fwd[i] != record.offset) {
-                syslog(LOG_ERR, "DBERROR: twoskip broken linkage %s: %08llX at %d, expected %08llX",
-                       FNAME(db), (LLU)record.offset, i, (LLU)fwd[i]);
+                /* XXX semantically, what is "i" here? */
```
Comment (ksmurchison) [inline lib/cyrusdb_twoskip.c:2121]: Looks like "i" is record.level

File: lib/cyrusdb_twoskip.c [hunk 2]
```diff
@@ -2126,17 +2136,20 @@ static int myconsistent(struct dbengine *db, struct txn *tid)
 
     for (i = 0; i < MAXLEVEL; i++) {
         if (fwd[i]) {
-            syslog(LOG_ERR, "DBERROR: twoskip broken tail %s: %08llX at %d",
-                   FNAME(db), (LLU)fwd[i], i);
+            /* XXX is "offset" correct? */
```
Comment (ksmurchison) [inline lib/cyrusdb_twoskip.c:2139]: I think "i" is level again

**Review by ksmurchison — APPROVED** [review]
Compiles cleanly with GCC 10.2.1

File: lib/cyrusdb_twoskip.c [hunk 3]
```diff
@@ -2111,8 +2118,11 @@ static int myconsistent(struct dbengine *db, struct txn *tid)
         for (i = 0; i < record.level; i++) {
             /* check the old pointer was to here */
             if (fwd[i] != record.offset) {
-                syslog(LOG_ERR, "DBERROR: twoskip broken linkage %s: %08llX at %d, expected %08llX",
-                       FNAME(db), (LLU)record.offset, i, (LLU)fwd[i]);
+                /* XXX semantically, what is "i" here? */
+                xsyslog(LOG_ERR, "DBERROR: twoskip broken linkage",
+                                 "filename=<%s> offset=<%08llX> i=<%d>"
+                                 " expected=<%08llX>",
+                                 FNAME(db), (LLU)record.offset, i, (LLU)fwd[i]);
```
Comment (brong) [inline lib/cyrusdb_twoskip.c:2125]: Yep, as Ken said - it should be `level=<%d>`  ... `i`.

File: lib/cyrusdb_twoskip.c [hunk 4]
```diff
@@ -2126,17 +2136,20 @@ static int myconsistent(struct dbengine *db, struct txn *tid)
 
     for (i = 0; i < MAXLEVEL; i++) {
         if (fwd[i]) {
-            syslog(LOG_ERR, "DBERROR: twoskip broken tail %s: %08llX at %d",
-                   FNAME(db), (LLU)fwd[i], i);
+            /* XXX is "offset" correct? */
+            xsyslog(LOG_ERR, "DBERROR: twoskip broken tail",
+                             "filename=<%s> offset=<%08llX> i=<%d>",
+                             FNAME(db), (LLU)fwd[i], i);
```
Comment (brong) [inline lib/cyrusdb_twoskip.c:2142]: Ditto, `level=`

**Review by brong — CHANGES_REQUESTED** [review]
Just changing 'i' to level will be good - otherwise looks great

**Comment by elliefm** [comment]
Updated those `i=<%d>`'s to `level=<%d>`

**Pull request merged**